testhelpers/core.go

```diff
@@ -184,3 +187,150 @@ func RequireCreateStorages(ctx context.Context, t *testing.T) (state.Tree, vm.St
 
 	return st, vms
 }
+
+type testStorage struct {
+	state interface{}
+}
+
+var _ exec.Storage = testStorage{}
+
+func (ts testStorage) Put(v interface{}) (cid.Cid, error) {
+	return cid.Cid{}, nil
+
+}
+func (ts testStorage) Get(cid.Cid) ([]byte, error) {
+	node, err := cbor.WrapObject(ts.state, types.DefaultHashFunction, -1)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return node.RawData(), nil
+}
+
+func (ts testStorage) Commit(cid.Cid, cid.Cid) error {
+	return nil
+}
+
+func (ts testStorage) Head() cid.Cid {
+	return cid.Cid{}
+}
+
+// FakeVMContext creates the scaffold for faking out the vm context for direct calls to actors
+type FakeVMContext struct {
+	TestMessage     *types.Message
+	TestStorage     exec.Storage
+	TestBalance     types.AttoFIL
+	TestBlockHeight *types.BlockHeight
+	TestVerifier    exec.Verifier
+	TestRandomness  []byte
+	Sender          func(to address.Address, method string, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
+	Addresser       func() (address.Address, error)
+	ActorTyper      func() bool
+	Charger         func(cost types.GasUnits) error
+	Sampler         func(sampleHeight *types.BlockHeight) ([]byte, error)
+	ActorCreator    func(addr address.Address, code cid.Cid, initalizationParams interface{}) error
+}
+
+var _ exec.VMContext = &FakeVMContext{}
+
+// NewFakeVMContext fakes the state machine infrastructure so actor methods can be called directly
+func NewFakeVMContext(message *types.Message, state interface{}) *FakeVMContext {
+	randomness := MakeRandomBytes(32)
+	return &FakeVMContext{
+		TestMessage:     message,
+		TestStorage:     &testStorage{state: state},
+		TestBlockHeight: types.NewBlockHeight(0),
+		TestBalance:     types.ZeroAttoFIL,
+		TestRandomness:  randomness,
+		Charger: func(cost types.GasUnits) error {
+			return nil
+		},
+		Sampler: func(sampleHeight *types.BlockHeight) ([]byte, error) {
+			return randomness, nil
+		},
+		Sender: func(to address.Address, method string, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error) {
+			return [][]byte{}, 0, nil
+		},
```

Comment:
What is the reason to exclude the addressor, actorTyper, actorCreater and verifier from the default setup?  Is it ok that methods on the output of `NewFakeVMContext` panic when called?

Reply:
pure, unadulterated laziness